scalding-core/src/main/scala/com/twitter/scalding/ExecutionContext.scala

```diff
@@ -31,6 +36,25 @@ trait ExecutionContext {
   def flowDef: FlowDef
   def mode: Mode
 
+  def getIdentifierOpt(descriptions: Seq[String]): Option[String] = {
+    if (descriptions.nonEmpty) {
+      Some(descriptions.distinct.mkString(", "))
+    } else {
+      None
+    }
+  }
+
+  def updateFlowStepName(step: BaseFlowStep[JobConf], descriptions: Seq[String]): BaseFlowStep[JobConf] = {
+    getIdentifierOpt(descriptions).foreach(newIdentifier => {
+      val stepXofYData = """\(\d+/\d+\)""".r.findFirstIn(step.getName).getOrElse("")
+      // Reflection is only temporary.  Latest cascading has setName public: https://github.com/cwensel/cascading/commit/487a6e9ef#diff-0feab84bc8832b2a39312dbd208e3e69L175
```

Comment:
I'll also add it in the comment.
https://github.com/twitter/scalding/issues/1294